### PR TITLE
feat: aggregation

### PR DESCRIPTION
**Pull request by vitaligi**
* implement aggregation
* refactor polygon parts

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/aggregation/models/aggregationManager.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    const findOptions = { id: aggregationParams.catalogId };
+    const layerMetadatas = await this.catalogClient.find(findOptions);
+    const layerMetadata = layerMetadatas.at(0);
+
+    if (layerMetadata === undefined) {
+      throw new NotFoundError('Could not find a catalog layer for the requested id');
+    }
+
+    if (layerMetadata.metadata?.productId === undefined) {
```
Comment (CL-SHLOMIKONCHA) [inline src/aggregation/models/aggregationManager.ts:32]: Implement Zod schema for find at the service ATM and replace the fields validations

File: src/aggregation/models/aggregationManager.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+      if (aggregation === undefined) {
+        throw new InternalServerError('Could not calculate aggregation');
+      }
+
+      return aggregation;
+    } catch (error) {
+      const errorMessage = `Could not aggregate polygon parts`;
+      this.logger.error({ msg: errorMessage, error });
+      throw new InternalServerError(errorMessage);
```
Comment (CL-SHLOMIKONCHA) [inline src/aggregation/models/aggregationManager.ts:58]:
```suggestion
      throw error;
```

File: src/polygonParts/models/polygonPartsManager.ts
Comment (CL-SHLOMIKONCHA) [inline src/polygonParts/models/polygonPartsManager.ts:1]: remove unused imports

File: tests/integration/polygonParts/polygonParts.spec.ts
Comment (CL-SHLOMIKONCHA) [inline tests/integration/polygonParts/polygonParts.spec.ts:1]: remove unused import

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: config/custom-environment-variables.json [hunk 3]
```diff
@@ -39,6 +39,21 @@
...
+      "__name": "HTTP_RETRY_RESET_TIMEOUT",
+      "__format": "boolean"
+    }
+  },
+  "disableHttpClientLogs": {
+    "__name": "DISABLE_HTTP_CLIENT_LOGS",
+    "__format": "boolean"
+  },
+  "servicesUrl": {
+    "catalogManager": "CATALOG_MANAGER_URL"
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply config/custom-environment-variables.json:55]: i kept the same naming convention as in overseer, is there a reason to change? it should also affect the helm configmap

File: helm/values.yaml [hunk 4]
```diff
@@ -59,6 +63,11 @@ env:
       polygonParts:
         namePrefix: ''
         nameSuffix: ''
+  httpRetry:
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply helm/values.yaml:66]: it's under the `env` hierarchy

File: openapi3.yaml [hunk 5]
```diff
@@ -10,6 +10,39 @@ tags:
...
+      tags:
+        - aggregation
+      summary: Get aggregated polygon-parts data by catalog id
+      description: Get aggregated polygon-parts data by catalog id
+      operationId: getAggregationMetadata
+      parameters:
+        - in: path
+          name: catalogId
+          schema:
+            type: string
```
[the remark this answers is not in the document]
Comment (vitaligi) [reply openapi3.yaml:24]: adding

File: src/polygonParts/models/polygonPartsManager.ts
Comment (vitaligi) [reply src/polygonParts/models/polygonPartsManager.ts:1]: removing

File: src/aggregation/models/aggregationManager.ts (hunk 2 above)
Comment (vitaligi) [reply src/aggregation/models/aggregationManager.ts:58]: applied

File: src/aggregation/models/aggregationManager.ts (hunk 1 above)
Comment (vitaligi) [reply src/aggregation/models/aggregationManager.ts]: added
we will add them from external source later

File: src/aggregation/models/validations.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { VALIDATIONS, type AggregationLayerMetadata } from '@map-colonies/mc-model-types';
+import type { MultiPolygon, Polygon } from 'geojson';
+import { z, type ZodType } from 'zod';
+
+export const aggregationMetadataSchema: ZodType<AggregationLayerMetadata> = z.object(
```
Comment (CL-SHLOMIKONCHA) [inline src/aggregation/models/validations.ts:5]: move to mc-models in order to support reusing

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: src/aggregation/models/validations.ts (hunk 6 above)
Comment (vitaligi) [reply src/aggregation/models/validations.ts]: done

**Pull request merged**